### PR TITLE
Feature/initial attempt

### PR DESCRIPTION
**Pull request by AdamUdovich**
Diff for rolling compression.

### CODE REVIEW

File: src/Serilog.Sinks.File/Sinks/File/RollingFileSink.cs [hunk 1]
```diff
@@ -132,13 +132,10 @@ public void GZipCompress(string logFile, string logDirectory)
...
-                        var buffer = new byte[1024];
-                        int bytesRead;
+                        int bufferSize = 1024;
+                        var buffer = new byte[bufferSize];
 
-                        while ((bytesRead = inputStream.Read(buffer, 0, buffer.Length)) > 0)
-                        {
-                            gzipStream.Write(buffer, 0, bytesRead);
-                        }
+                        inputStream.CopyTo(gzipStream, bufferSize);
```
Comment (cocowalla) [inline src/Serilog.Sinks.File/Sinks/File/RollingFileSink.cs]: Such a small buffer size is going to result in sub-par compression ratios - what's the purpose of overriding the default buffer size here? (I guess the default is 64K?)

File: src/Serilog.Sinks.File/Sinks/File/RollingFileSink.cs [hunk 2]
```diff
@@ -132,13 +132,10 @@ public void GZipCompress(string logFile, string logDirectory)
                 {
                     using (var gzipStream = new System.IO.Compression.GZipStream(outputStream, System.IO.Compression.CompressionMode.Compress))
                     {
-                        var buffer = new byte[1024];
-                        int bytesRead;
+                        int bufferSize = 1024;
+                        var buffer = new byte[bufferSize];
```
Comment (cocowalla) [inline src/Serilog.Sinks.File/Sinks/File/RollingFileSink.cs]: I don't think `buffer` is being used?

File: src/Serilog.Sinks.File/Sinks/File/RollingFileSink.cs [hunk 3]
```diff
@@ -174,7 +190,8 @@ void AlignCurrentFileTo(DateTime now, bool nextSequence = false)
                     {
                         var directoryFileName = Path.GetFileName(directoryFile);
 
-                        if (!(directoryFileName.Contains("-GZip")))
+                        // Not sure if we can assume log files will always be .txt?
```
Comment (cocowalla) [inline src/Serilog.Sinks.File/Sinks/File/RollingFileSink.cs]: Nope :smile:

The filename (and extension) are configurable by the user

File: src/Serilog.Sinks.File/Sinks/File/RollingFileSink.cs [hunk 4]
```diff
@@ -87,39 +91,51 @@ public void Compress(string logFile, string logDirectory, CompressionType compre
...
         }
 
-        public void GZipCompress(string prevLog, string logDirectory)
+        /// <summary>
+        /// Uses GZip compression algorithm to compress logFile into a .gzip file
+        /// The GZip algorithm is from System.IO.Compression
+        /// </summary>
+        /// <param name="logFile"> The log file to be compressed </param>
+        /// <param name="logDirectory"> The directory that holds logFile </param>
+        public void GZipCompress(string logFile, string logDirectory)
```
Comment (cocowalla) [inline src/Serilog.Sinks.File/Sinks/File/RollingFileSink.cs]: Why `public`?

**Comment by nblumhardt** [comment]
@AdamUdovich I think #80 is going to enable this; I'm catching up and doing a bit of housekeeping, so I'll close the PR here - kudos for pushing it forwards :-)

**Pull request closed without merging**